### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by davidsulc**
Made a few changes to the documentation in the hopes it will facilitate onboardig of new users.

I hope these aren't perceived as nit-picking: they're relatively trivial changes, but are things that either caught me off guard (args map key stringification), or had me scratching my head (why the queue was specified as a string in the worker, when it'specified as an atom in the config).

### CODE REVIEW

File: lib/oban/worker.ex [hunk 1]
```diff
@@ -173,6 +173,9 @@ defmodule Oban.Worker do
   @doc """
   The `perform/2` function is called when the job is executed.
 
+  The map given as the first argument will have string keys, even if atom keys were used when
```
Comment (sorentwo) [inline lib/oban/worker.ex]: As mentioned above, this is only true when Oban is executing jobs, not if you call `MyApp.Worker.perform/2` directly.

**Comment by davidsulc** [comment]
I've reproduced the README changes within the `Oban` module docs.

I don't know if it would be of interest to you, but here's a neat trick to ensure they remain consistent: https://github.com/plataformatec/nimble_parsec/blob/master/lib/nimble_parsec.ex#L2

File: README.md [hunk 2]
```diff
@@ -248,15 +248,17 @@ concurrently. Here are a few caveats and guidelines:
 
 Worker modules do the work of processing a job. At a minimum they must define a
 `perform/2` function, which is called with an `args` map and the job struct.
+The keys in the `args` map will always be converted to strings prior to calling
```
[the remark this answers is not in the document]
Comment (davidsulc) [reply README.md]: Good point, I've clarified the change to indicate that the string keys happen when Oban is calling `perform/2`

File: README.md [hunk 3]
```diff
@@ -248,15 +248,17 @@ concurrently. Here are a few caveats and guidelines:
...
 Define a worker to process jobs in the `events` queue:
 
 ```elixir
 defmodule MyApp.Business do
-  use Oban.Worker, queue: "events", max_attempts: 10
+  use Oban.Worker, queue: :events, max_attempts: 10
 
   @impl Oban.Worker
-  def perform(%{"id" => id}, _job) do
+  def perform(%{"id" => id, industry: _industry}, _job) do
```
[the remark this answers is not in the document]
Comment (davidsulc) [reply README.md]: Oops, mistake on my part: I've removed the `:industry` key per your other comments

File: README.md [hunk 4]
```diff
@@ -284,16 +286,16 @@ database. For convenience and consistency all workers provide a `new/2`
 function that converts an args map into a job changeset suitable for insertion:
 
 ```elixir
-%{in_the: "business", of_doing: "business"}
+%{id: 1, industry: "technology", in_the: "business", of_doing: "business"}
```
[the remark this answers is not in the document]
Comment (davidsulc) [reply README.md]: My idea was to logically link the examples in the "Enqueueing Jobs" section with the one in "Defining Workers". In the original docs, it's not explicit that the args map used to enqueue the job is made fully available to `perform/2`.
I've tried to showcase this with e952f1bf76edc0cb7ffb1fcded8b9d0222b9df41 : let me know if that works better or should just be dropped.

**Comment by sorentwo** [comment]
@davidsulc Sorry for the delay, I got sidetracked by the queue performance work. Thanks for the updates and the README trick from nimble parsec, I'm taking that =)

**Pull request merged**